Routes. Option to lower route priority

iRedds opened the pull request:
**Description**
Work with modules was added in CI4, but the following issue is not resolved.
A situation may arise when an application route containing a wildcard will process routes specified in the module. 

Example 
```php
// App\Config\Routes.php
$routes->add('(.*)', 'MainController::wildcard');

// Modules\Acme\Config\Routes.php
$routes->add('admin', 'AdminController::index');
```
In this case, the URL __/admin__ will be handled by the route __$routes->add('(. *)', 'MainController :: wildcard');__

This PR adds an option that can lower the priority of the route. 

```php
// App\Config\Routes.php
$routes->add('(.*)', 'MainController::wildcard', ['priority' => 1]);

// Modules\Acme\Config\Routes.php
$routes->add('admin', 'AdminController::index');
```
Now the URL __/admin__ will be handled by the route __$routes->add('admin', 'AdminController::index');__

Changes:
* Added "priority" option to lower route priority.
* Sorting by priority is disabled by default.
* ~~Added methods to RouteCollectionInterface: RouteCollectionInterface::enablePrioritySorting() and RouteCollectionInterface::isPrioritySortingEnabled().~~

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPdocs
- [x] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

MGatner reviewed (changes requested):
We cannot modify the interface without a major version release. I like this change but it will have to be handled differently. For example, you could check for `method_exists()` in `Router` and thus make it an "optional" method.

Some opinions: I would also be in favor of using set/get methods since this is the style of the rest of framework. Something like `protected $prioritize`, `setPrioritize()`, and `getPrioritize()`. I think using "priority" instead of "order" in the collections would be more consistent and make more sense as long as the docs were clear about what higher/lower numbers means.

MGatner commented:
I meant to leave a comment before the review, but I think this is a very handy feature! And a nice PR, notes aside.

iRedds commented:
@MGatner I want to know your opinion.
What if you remove the suggested methods altogether and sort always?

iRedds commented:
* Removed methods from RouteCollectionInterface.
* Added RouteCollection::setPrioritize() method.
* Sorting moved from Router class to RouteCollection.
* Added detection of the "priority" option when adding a route.

iRedds commented on user_guide_src/source/incoming/routing.rst:
```diff
@@ -427,6 +427,31 @@ be used when the first parameter is a language string::
+
+
+.. note:: By default, all routes have a priority of 0.
+    The higher the number specified in "order", the lower the priority of the route.
```
According to my idea, the priority is the name of the functional.
That is, the option specifies the order in the queue and the priority for processing is built on the basis of the queue. The higher the queue value, the lower the priority for processing. 
Probably influenced by @MGatner's comment, I decided that the name "priority" would be more suitable for understanding how this functionality works. 
Maybe I was wrong?

paulbalandan replied:
If the original idea for the number is queueing, then it's right that the higher the number the lower the running. So the option should be aptly named to that, i.e. `queue`.

However, if we'll go with the `priority` option key instead, it'd be understood that higher the number the higher the running.

paulbalandan replied:
Ok, I just got a realization. In retrospect, what you're saying is right. Forget all of these. Forgive my confusion. 😂

iRedds replied:
Aaaa !!! So what should I choose as the option name? Order, queue, or priority?
Please don't do like @MGatner.
1. Better to do this and that.
2. In general, everything is cool. Forget what I said. 
:sob:

paulbalandan replied:
Sorry for the confusion. Current is okay with me. Just change the wordings here in the doc from order to priority.

iRedds commented on system/Router/RouteCollection.php:
```diff
@@ -1367,6 +1397,17 @@ protected function create(string $verb, string $from, $to, array $options = null
+		{
+			$options['priority'] = abs((int) $options['priority']);
+
+			if ($options['priority'] > 0)
```
Here, an actually negative number will be cast to an absolute value. ``-1 -> 1``

paulbalandan replied:
Ah yes. Forgot the `abs()`. Anyway, pls add a test case for `0` priority.

Also, we need to document that negative priorities will be resolved to absolute values. People might be thinking why priority -1 runs aligned with priority 1.

paulbalandan commented:
@MGatner how about this one?

MGatner reviewed (approved):
This looks good. I would like to highlight this in our next release notes - that can be a part of this PR or added separately.

iRedds commented:
@MGatner note added
